iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -92,379 +77,376 @@ TEST_F(ServiceDiscovery_test, FindServiceWithWildcardsReturnsOnlyIntrospectionSe
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION2));
 }
 
-TEST_F(ServiceDiscovery_test, SubscribeAnyInstance)
+TEST_F(ServiceDiscovery_test, FindServiceWithInstanceWildcardReturnsAllMatchingServices)
 {
     ::testing::Test::RecordProperty("TEST_ID", "6e0b1a12-6995-45f4-8fd8-59acbca9bfa8");
-    EXPECT_TRUE(sut.offerService({"service1", "instance1", "event1"}));
-    EXPECT_TRUE(sut.offerService({"service1", "instance2", "event2"}));
-    EXPECT_TRUE(sut.offerService({"service1", "instance3", "event3"}));
+
+    const IdString_t SERVICE = "service";
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION1(SERVICE, "instance1", "event1");
+    iox::popo::UntypedPublisher publisher_sd1(SERVICE_DESCRIPTION1);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION2(SERVICE, "instance2", "event2");
+    iox::popo::UntypedPublisher publisher_sd2(SERVICE_DESCRIPTION2);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION3(SERVICE, "instance3", "event3");
+    iox::popo::UntypedPublisher publisher_sd3(SERVICE_DESCRIPTION3);
+
     this->InterOpWait();
+
     ServiceContainer serviceContainerExp;
-    serviceContainerExp.push_back({"service1", "instance1", "event1"});
-    serviceContainerExp.push_back({"service1", "instance2", "event2"});
-    serviceContainerExp.push_back({"service1", "instance3", "event3"});
+    serviceContainerExp.push_back(SERVICE_DESCRIPTION1);
+    serviceContainerExp.push_back(SERVICE_DESCRIPTION2);
+    serviceContainerExp.push_back(SERVICE_DESCRIPTION3);
 
-    auto serviceContainer = sut.findService(IdString_t("service1"), iox::runtime::Wildcard_t());
+    auto serviceContainer = sut.findService(SERVICE, iox::runtime::Wildcard_t());
     ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(3u));
+    ASSERT_THAT(serviceContainer.value().size(), Eq(3U));
     EXPECT_TRUE(serviceContainer.value() == serviceContainerExp);
 }
 
-TEST_F(ServiceDiscovery_test, OfferSingleMethodServiceMultiInstance)
+TEST_F(ServiceDiscovery_test, FindServiceWithServiceWildcardReturnsAllServices)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "538bec69-ea02-400e-8643-c833d6e84972");
-    EXPECT_TRUE(sut.offerService({"service1", "instance1", "event1"}));
-    EXPECT_TRUE(sut.offerService({"service1", "instance2", "event2"}));
-    EXPECT_TRUE(sut.offerService({"service1", "instance3", "event3"}));
-    this->InterOpWait();
+    ::testing::Test::RecordProperty("TEST_ID", "f4731a52-39d8-4d49-b247-008a2e9181f9");
+    const IdString_t INSTANCE = "instance";
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION1("service1", INSTANCE, "event1");
+    iox::popo::UntypedPublisher publisher_sd1(SERVICE_DESCRIPTION1);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION2("service2", "INSTANCE", "event2");
```

Comment:
Are you sure you want `"INSTANCE"`? If yes, please give it another name since it looks like a mistake.

Reply:
Yes, I wanted another instance id string. After copying the lines above, it was easy to just add quotation marks. But sure, it could also be `"Horst"`. I'll change this.